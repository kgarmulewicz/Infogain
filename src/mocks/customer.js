export const customer = {
            "id": 1,
            "firstName": "Loralee",
            "lastName": "Timeby",
            "transactions": [
                {
                    "date": "2023-01-01T00:00:00.000Z",
                    "amount": 276
                },
                {
                    "date": "2023-01-04T00:00:00.000Z",
                    "amount": 1
                },
                {
                    "date": "2023-01-06T00:00:00.000Z",
                    "amount": 95
                },
                {
                    "date": "2023-01-09T00:00:00.000Z",
                    "amount": 209
                },
                {
                    "date": "2023-01-13T00:00:00.000Z",
                    "amount": 134
                },
                {
                    "date": "2023-01-15T00:00:00.000Z",
                    "amount": 174
                },
                {
                    "date": "2023-01-20T00:00:00.000Z",
                    "amount": 231
                },
                {
                    "date": "2023-01-21T00:00:00.000Z",
                    "amount": 246
                },
                {
                    "date": "2023-01-24T00:00:00.000Z",
                    "amount": 177
                },
                {
                    "date": "2023-01-31T00:00:00.000Z",
                    "amount": 128
                },
                {
                    "date": "2023-02-05T00:00:00.000Z",
                    "amount": 142
                },
                {
                    "date": "2023-02-11T00:00:00.000Z",
                    "amount": 191
                },
                {
                    "date": "2023-02-18T00:00:00.000Z",
                    "amount": 256
                },
                {
                    "date": "2023-02-24T00:00:00.000Z",
                    "amount": 67
                },
                {
                    "date": "2023-02-25T00:00:00.000Z",
                    "amount": 75
                },
                {
                    "date": "2023-02-28T00:00:00.000Z",
                    "amount": 130
                },
                {
                    "date": "2023-03-06T00:00:00.000Z",
                    "amount": 79
                },
                {
                    "date": "2023-03-11T00:00:00.000Z",
                    "amount": 186
                },
                {
                    "date": "2023-03-14T00:00:00.000Z",
                    "amount": 44
                },
                {
                    "date": "2023-03-16T00:00:00.000Z",
                    "amount": 209
                },
                {
                    "date": "2023-03-23T00:00:00.000Z",
                    "amount": 88
                },
                {
                    "date": "2023-03-26T00:00:00.000Z",
                    "amount": 215
                },
                {
                    "date": "2023-03-28T23:00:00.000Z",
                    "amount": 152
                },
                {
                    "date": "2023-03-30T23:00:00.000Z",
                    "amount": 279
                }
            ]
        };