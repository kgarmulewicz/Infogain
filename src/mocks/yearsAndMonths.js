export const yearsAndMonths = [
    {
        "year": 2023,
        "month": 1
    },
    {
        "year": 2023,
        "month": 2
    },
    {
        "year": 2023,
        "month": 3
    }
]