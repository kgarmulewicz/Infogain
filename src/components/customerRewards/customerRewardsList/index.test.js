import '@testing-library/jest-dom/extend-expect'
import React from 'react'
import {render, screen} from '@testing-library/react'
import {customer} from '../../../mocks/customer';
import CustomerRewardsList from "./index";

describe("<CustomerRewardsList />", () => {
    const props = {
        customer 
    }
test('CustomerRewardsList match snapshot!', () => {
  const {view} = render(<CustomerRewardsList {...props} />)
  expect(view).toMatchSnapshot();
})
})

describe("CustomerRewardsList Component", () => {
    const transactions = [
      { date: "2021-01-01", amount: 100 },
      { date: "2021-02-01", amount: 200 },
      { date: "2021-03-01", amount: 300 }
    ];
  
    it("renders a table with correct number of headers", () => {
      render(<CustomerRewardsList customer={{ transactions }} />);
      const headers = screen.getAllByTestId('transactions-list-header');
      expect(headers.length).toBe(3);
    });
  
    it("renders the correct number of rows", () => {
      render(<CustomerRewardsList customer={{ transactions }} />);
      const rows = screen.getAllByTestId("transactions-list-row");
      expect(rows.length).toBe(3);
    });
  });
