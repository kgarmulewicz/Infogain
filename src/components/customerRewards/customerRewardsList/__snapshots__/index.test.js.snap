// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CustomerRewardsList /> CustomerRewardsList match snapshot! 1`] = `undefined`;
